model/flow/chunk.go

```diff
@@ -194,18 +222,35 @@ func NewChunkDataPack(untrusted UntrustedChunkDataPack) (*ChunkDataPack, error)
 		return nil, fmt.Errorf("ExecutionDataRoot.ChunkExecutionDataIDs must not be empty")
 	}
 
+	return NewChunkDataPack(
+		untrusted.ChunkID,
+		untrusted.StartState,
+		untrusted.Proof,
+		untrusted.Collection,
+		untrusted.ExecutionDataRoot,
+	), nil
+}
+
+func NewChunkDataPack(chunkID Identifier, startState StateCommitment, proof StorageProof, collection *Collection, executionDataRoot BlockExecutionDataRoot) *ChunkDataPack {
```

Comment:
I think `FromUntrustedChunkDataPack` should be the exported constructor (probably renamed to `NewChunkDataPack`) and what is now `NewChunkDataPack` should become private. 

Using the constructor is negligible overhead and there is only a single place [here](https://github.com/onflow/flow-go/blob/b89fb526a68721ebc5773700636c092d2e3ab2e3/storage/store/chunk_data_packs.go#L282) where marginal changes would be required. Instantiating an `UntrustedChunkDataPack` first is a bit more boiler plate code, but we found it to be relatively limited extra work while providing broad conceptual correctness in the code base.